app/src/main/java/org/mozilla/focus/settings/BaseSettingsComposeFragment.kt

```diff
@@ -12,65 +16,78 @@ import androidx.compose.material.Text
+import androidx.compose.runtime.CompositionLocalProvider
 import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.res.colorResource
 import androidx.compose.ui.res.stringResource
 import androidx.fragment.app.Fragment
 import com.google.accompanist.insets.LocalWindowInsets
-import com.google.accompanist.insets.ProvideWindowInsets
 import com.google.accompanist.insets.rememberInsetsPaddingValues
 import com.google.accompanist.insets.ui.TopAppBar
 import org.mozilla.focus.R
 import org.mozilla.focus.ui.theme.FocusTheme
 import org.mozilla.focus.ui.theme.focusColors
+import org.mozilla.focus.utils.StatusBarUtils
 
-open class BaseSettingsComposeFragment : Fragment() {
+/**
+ * BaseSettingsComposeFragment displays a toolbar in compose for the screen
+ */
+abstract class BaseSettingsComposeFragment : Fragment() {
+    /**
+     * title of the screen.
+     */
+    abstract fun title(): String
+
+    /**
+     * onClickUpButton callback for the back arrow of the toolbar
+     */
+    abstract fun onClickUpButton(): () -> Unit
 
     /**
-     * Displays a toolbar for the screen
-     * @param title of the screen.
-     * @param onClickUpButton callback for the back arrow of the toolbar
-     * @param content of the screen in compose. That will be shown below Toolbar
+     * content of the screen in compose. That will be shown below Toolbar
      */
     @Composable
-    fun toolbar(
-        title: String,
-        onClickUpButton: () -> Unit,
-        content: @Composable () -> Unit
-    ): ComposeView {
-        return ComposeView(requireContext()).apply {
-            FocusTheme {
-                Scaffold {
-                    Column {
-                        ProvideWindowInsets {
-                            TopAppBar(
-                                title = {
-                                    Text(
-                                        text = title,
-                                        color = focusColors.toolbarColor
-                                    )
-                                },
-                                contentPadding = rememberInsetsPaddingValues(
-                                    insets = LocalWindowInsets.current.statusBars,
-                                    applyStart = true,
-                                    applyTop = true,
-                                    applyEnd = true,
-                                ),
-                                navigationIcon = {
-                                    IconButton(
-                                        onClick = onClickUpButton
-                                    ) {
-                                        Icon(
-                                            Icons.Filled.ArrowBack,
-                                            stringResource(R.string.go_back),
-                                            tint = focusColors.toolbarColor
+    abstract fun ContentScreen()
```

Comment:
nit: Maybe just `Content`?

Reply:
Doesn't work with Content() because Content method from Compose Class will get called and the app will crash

Reply:
I think the pattern in the API is to have the content as a composable function with a return type. 
So we could name it `content()` with a  `() -> Unit` return type.

Reply:
I had a comment that went missing 😮 .
You can either: 
1. keep the `Content` naming and use `this@BaseComposeFragment.Content()` to avoid calling the `ComposeView` method (that is happening because the calling place is in the `ComposeView` scope)
2. Use the API pattern to have it as a `val` with a `@Composable ()-> Type` that you can simply name `content`. (This will involve changing the way you overrode the content.